salt/modules/useradd.py

```diff
@@ -60,28 +60,30 @@ def _get_gecos(name):
     Retrieve GECOS field info and return it in dictionary form
     '''
     gecos_field = salt.utils.stringutils.to_unicode(
-        pwd.getpwnam(_quote_username(name)).pw_gecos).split(',', 3)
+        pwd.getpwnam(_quote_username(name)).pw_gecos).split(',', 4)
     if not gecos_field:
         return {}
     else:
         # Assign empty strings for any unspecified trailing GECOS fields
-        while len(gecos_field) < 4:
+        while len(gecos_field) < 5:
             gecos_field.append('')
         return {'fullname': salt.utils.locales.sdecode(gecos_field[0]),
                 'roomnumber': salt.utils.locales.sdecode(gecos_field[1]),
                 'workphone': salt.utils.locales.sdecode(gecos_field[2]),
-                'homephone': salt.utils.locales.sdecode(gecos_field[3])}
+                'homephone': salt.utils.locales.sdecode(gecos_field[3]),
+                'other': salt.utils.locales.sdecode(gecos_field[4])}
 
 
 def _build_gecos(gecos_dict):
     '''
     Accepts a dictionary entry containing GECOS field names and their values,
     and returns a full GECOS comment string, to be used with usermod.
     '''
-    return '{0},{1},{2},{3}'.format(gecos_dict.get('fullname', ''),
-                                    gecos_dict.get('roomnumber', ''),
-                                    gecos_dict.get('workphone', ''),
-                                    gecos_dict.get('homephone', ''))
+    return '{0},{1},{2},{3},{4}'.format(gecos_dict.get('fullname', ''),
+                                        gecos_dict.get('roomnumber', ''),
+                                        gecos_dict.get('workphone', ''),
+                                        gecos_dict.get('homephone', ''),
+                                        gecos_dict.get('other', ''),).rstrip(',')
```

Comment:
I'd go here more pythonic way, actually:
```python
out = []
for field in ['fullname', 'roomnumber', 'workphone', 'homephone', 'other']:
    field = gecos_dict.get(field)
    if field:
        out.append(field)
return ','.join(out)
```

Reply:
I would prefer to keep the current logic which is simpler and explicit rather than replacing it with a for loop + more variables and some extra calls, which unnecessarily makes the logic more complex.